src/collective/cover/events.py

```diff
@@ -49,14 +51,19 @@ def update_link_integrity(obj, event):
     :type event:
     """
     refs = set()
-    for name, value in obj.data.items():
-        if isinstance(value, RichTextValue):
-            value = value.raw
-            if not value:
-                continue
-            links = extractLinks(value)
-            refs |= getObjectsFromLinks(obj.context, links)
-
+    uuid = obj.data.get('uuid', None)
+    if uuid is not None:
+        refs |= {uuidToObject(uuid)}
+    if IListTile.providedBy(obj):
+        uuids = obj.data.get('uuids', [])
+        for uuid in uuids:
+            refs |= {uuidToObject(uuid)}
+    elif IRichTextTile.providedBy(obj):
+        value = obj.data.get('text')
+        value = value.raw
+        links = extractLinks(value)
+        refs |= getObjectsFromLinks(obj.context, links)
+    # needs plone.app.referenceablebehavior enabled in cover content type
```

Comment:
@rodfersou did you reviewed this? https://github.com/collective/collective.cover/pull/579